### PR TITLE
Localisation

### PR DESCRIPTION
**Pull request by katendejericho5**
#### Summary of Changes (What does this PR do?)

- [x] Adds support for French.
- [x] Updates existing tests to cater for French translations.

#### Status of maturity (all need to be checked before merging):

- [ ] I've tested this locally
- [ ] I consider this code done
- [ ] This change ready to hit production in its current state
- [ ] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] I've included issue number in the "Closes #ISSUE-NUMBER" part of the "[What are the relevant tickets?](#what-are-the-relevant-tickets)" section to [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [ ] I've updated corresponding documentation for the changes in this PR.
- [ ] I have written unit and/or e2e tests for my change(s).

#### How should this be manually tested?

- Please include the steps to be done inorder to setup and test this PR.

#### What are the relevant tickets?

- Closes #<enter issue number here>
- [Jira_card_number]() (If exists)

### CODE REVIEW

File: mobile/lib/main.dart [hunk 1]
```diff
@@ -42,6 +44,16 @@ void main() async {
...
         title: 'AirQo',
         theme: customTheme(),
+        localizationsDelegates: const [
+          AppLocalizations.delegate,
+          GlobalMaterialLocalizations.delegate,
+          GlobalWidgetsLocalizations.delegate,
+          GlobalCupertinoLocalizations.delegate,
+        ],
+        supportedLocales: const [
+          Locale('en'), // English
```
Comment (katendejericho5) [inline mobile/lib/main.dart:54]: sections for supported languages

File: mobile/lib/models/enum_constants.dart [hunk 2]
```diff
@@ -213,6 +214,83 @@ enum AirQuality {
...
 
+  String getSearchNearbyLocationsText(BuildContext context) {
+    switch (this) {
+      case AirQuality.good:
+        return AppLocalizations.of(context)!.goodQualityAirAroundYou;
+      case AirQuality.moderate:
+        return AppLocalizations.of(context)!.moderateQualityAirAroundYou;
+      case AirQuality.ufsgs:
+        return AppLocalizations.of(context)!
+            .nearbyLocationsWithAirQualityUnhealthyForSensitiveGroups;
```
Comment (katendejericho5) [inline mobile/lib/models/enum_constants.dart:225]: AppLocalizations.of(context)!
            .nearby........... it used by flutter to replace with the right languages(locale)

File: mobile/README.md [hunk 3]
```diff
@@ -166,6 +166,7 @@ flutter build appbundle --obfuscate --split-debug-info=${PWD}/obfuscate
 flutter build ipa --obfuscate --split-debug-info=${PWD}/obfuscate
 flutter build appbundle [--analyze-size]
 flutter build ipa [--analyze-size]
+ flutter gen-l10n
```
[the remark this answers is not in the document]
Comment (katendejericho5) [reply mobile/README.md:169]: run flutter run and codegen takes place automatically. You should find generated files in ${FLUTTER_PROJECT}/.dart_tool/flutter_gen/gen_l10n....... Alternatively, you can also run flutter gen-l10n to generate the same files without running the app.....

File: mobile/l10n.yaml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+arb-dir: lib/l10n
```
Comment (katendejericho5) [inline mobile/l10n.yaml:1]: This file configures the localization tool

File: mobile/test/screens/insights/insights_widgets_test.dart [hunk 5]
```diff
@@ -108,6 +109,16 @@ void main() {
   testWidgets('Health tip widget', (tester) async {
     await tester.pumpWidget(
       MaterialApp(
+        localizationsDelegates: const [
+          GlobalMaterialLocalizations.delegate,
+          GlobalWidgetsLocalizations.delegate,
+          GlobalCupertinoLocalizations.delegate,
+        ],
+        supportedLocales: const [
+          Locale('en'),
```
Comment (NoahNsimbe) [inline mobile/test/screens/insights/insights_widgets_test.dart:118]: Updated tests to use localizations.

**Review by NoahNsimbe — APPROVED** [review]
LGTM Thanks @katendejericho5

**Review by Baalmart — APPROVED** [review]

**Pull request merged**